badger_voter_sdk/collectors/data_processors.py

```diff
@@ -37,3 +37,37 @@ def extract_voting_power_per_pool(
                 (Decimal(weight) / all_together) * Decimal(amount_of_vltoken_voted)
             )
     return pool_votes
+
+
+def extract_voting_power_per_eoa(
+        target_wallet: str, voters: Dict[str, Dict], scores: Dict
+) -> Optional[Dict[str, Decimal]]:
+    """
+    Extract voting power by wallet, combining two datasets:
+        1. voters - has weighed votes per wallet
+        2. scores - has data about vlTOKEN voted by wallet
+
+    Returns voting choices for wallet calculated in a form {"1": Decimal(123123.04), ...}
+    """
+    if not all([target_wallet, voters, scores]):
+        return
+    # First, map both dataset's wallets to have checksummed addresses
+    voters_checksummed = {
+        Web3.toChecksumAddress(wallet): choices for wallet, choices in voters.items()
+    }
+    scores_checksummed = {
+        Web3.toChecksumAddress(wallet): totals for wallet, totals in scores.items()
+    }
+    eoa_checksummed = Web3.toChecksumAddress(target_wallet)
+    amount_of_vltoken_voted = scores_checksummed.get(eoa_checksummed)
+    voter_choices = voters_checksummed.get(eoa_checksummed)
+    if not amount_of_vltoken_voted or not voters:
+        return
+    # All together - represents 100% as it's a vote weight
+    all_together = Decimal(sum(voter_choices.values()))
+    if all_together == Decimal(0):
+        return
+    return {
+        pool_choice: (Decimal(weight) / all_together) * Decimal(amount_of_vltoken_voted)
```

Comment:
Nit but could you define the values as Decimal when declaring the Dict rather than during comparison? Makes reading cleaner